packages/grafana-ui/src/components/uPlot/config/UPlotSeriesBuilder.ts

```diff
@@ -1,31 +1,267 @@
 import tinycolor from 'tinycolor2';
-import { Series } from 'uplot';
-import { PointMode } from '../config';
+import uPlot, { Series } from 'uplot';
+import { GraphMode, LineConfig, AreaConfig, PointsConfig, PointMode, LineInterpolation } from '../config';
 import { PlotConfigBuilder } from '../types';
 
-export interface SeriesProps {
+export interface SeriesProps extends LineConfig, AreaConfig, PointsConfig {
+  mode: GraphMode;
   scaleKey: string;
-  line?: boolean;
-  lineColor?: string;
-  lineWidth?: number;
-  points?: PointMode;
-  pointSize?: number;
-  pointColor?: string;
-  fill?: boolean;
-  fillOpacity?: number;
-  fillColor?: string;
+}
+
+function buildBarsPaths(u: uPlot, seriesIdx: number, idx0: number, idx1: number): Series.Paths {
```

Comment:
would be nice to move this part to a separate file (bars.ts or something). Preferably a unit test or two if possible.

Reply:
moved out to `paths.ts`